src/main/java/com/es/phoneshop/exception/OutOfStockException.java

```diff
@@ -0,0 +1,34 @@
+package com.es.phoneshop.exception;
+
+import com.es.phoneshop.model.product.Product;
+
+public class OutOfStockException extends Exception {
+
+    private final String MESSAGE = "Out of stock. Available stock is %d";
+
+    private Product product;
```

Comment:
You shouldn't store any business logic-related fields into the exception. Generally it is sufficient to have only message and root cause. In some specific situation it's also allowed to introduce new exception-specific field, e.g. errorCode, status, etc. If you can't resolve some issue and haven't any other choice, be sure at least that your Exception state is Immutable.